jsonschema.go

```diff
@@ -92,16 +96,33 @@ func checkString(rawdata interface{}, format interface{}) error {
 	return nil
 }
 
-func checkFloat(rawdata interface{}, format interface{}) error {
-	_, ok := rawdata.(float64)
+func checkNumber(rawdata interface{}) (json.Number, error) {
+	res, ok := rawdata.(json.Number)
 	if !ok {
-		return fmt.Errorf("expected float, got [%s]", reflect.TypeOf(rawdata))
+		return res, fmt.Errorf("expected json number, got [%s]", reflect.TypeOf(rawdata))
+	}
+	return res, nil
+}
+
+func checkFloat(rawdata interface{}, format interface{}) error {
+	number, err := checkNumber(rawdata)
+	if err != nil {
+		return err
+	}
+	if !strings.Contains(number.String(), ".") {
```

Comment:
Wont work, both conversions are always made then the value has no fractional part:

https://play.golang.org/p/XCoH-sCzRk

Perhaps if we are willing to allow numbers without fractional part to be considered float too, them it would be a good idea.

@matheusvill @kamilash what you think ?